searchlib/src/tests/attribute/searchable/attribute_searchable_adapter_test.cpp

```diff
@@ -472,6 +464,26 @@ TEST("require that direct attribute iterators work") {
     }
 }
 
+TEST("require that single weighted set turns filter on filter fields") {
+        bool fast_search = true;
+        bool strict = true;
+        bool isFilter = true;
+        MyAttributeManager attribute_manager = make_weighted_string_attribute_manager(fast_search, isFilter);
+        SimpleStringTerm empty_node("notfoo", "", 0, Weight(1));
+        Result empty_result = do_search(attribute_manager, empty_node, strict);
+        EXPECT_EQUAL(0u, empty_result.hits.size());
+        SimpleStringTerm node("foo", "", 0, Weight(1));
+        Result result = do_search(attribute_manager, node, strict);
+        EXPECT_EQUAL(3u, result.est_hits);
+        EXPECT_TRUE(result.iterator_dump.find("DocumentWeightSearchIterator") == vespalib::string::npos);
```

Comment:
how does not using the DocumentWeightSearchIterator test that it supports termwise evaluation?

Reply:
It does not test that it supports termwise, it just tests that it instantiates a filter iterator.

Reply:
but it is the 'weighted set blueprint' and not the 'document weight iterator' that is tested? (hence my confusion)